### PR TITLE
CASSANDRA-15857 Preserve frozen in RawTuple toString

### PR DESCRIPTION
**Pull request by yifan-c**

### CODE REVIEW

File: src/java/org/apache/cassandra/cql3/CQL3Type.java [hunk 1]
```diff
@@ -833,11 +835,7 @@ public boolean supportsFreezing()
...
             {
-                List<CQL3Type.Raw> frozenTypes =
-                    types.stream()
-                         .map(t -> t.supportsFreezing() ? t.freeze() : t)
-                         .collect(toList());
-                return new RawTuple(frozenTypes, true);
+                return this;
             }
 
             public CQL3Type prepare(String keyspace, Types udts) throws InvalidRequestException
```
Comment (snazy) [inline src/java/org/apache/cassandra/cql3/CQL3Type.java:841]: Think you can simplify the `frozen` check in the first statement

File: test/unit/org/apache/cassandra/io/sstable/CQLSSTableWriterTest.java [hunk 2]
```diff
@@ -66,24 +74,28 @@ public static void setup() throws Exception
         StorageService.instance.initServer();
     }
 
+    @Before
+    public void perTestSetup()
+    {
+        keyspace = "cql_keyspace" + idGen.incrementAndGet();
+        table = "table" + idGen.incrementAndGet();
+        qualifiedTable = keyspace + '.' + table;
+        File tempdir = Files.createTempDir();
```
Comment (snazy) [inline test/unit/org/apache/cassandra/io/sstable/CQLSSTableWriterTest.java]: Think it's nicer to use a JUnit 4 `@Rule` here for `tempdir`:
``` 
    @Rule
    private TemporaryFolder tempFolder = new TemporaryFolder();
```

Comment (snazy) [reply test/unit/org/apache/cassandra/io/sstable/CQLSSTableWriterTest.java]: Otherwise a nice simplification!

**Comment by snazy** [comment]
`DescribeStatementTest` has 2 new failures in `testDescribeFunctionAndAggregate` (line 132) + `testDescribe` (line 298)

**Comment by snazy** [comment]
Oh, and it seems that the `CQL3Type.Raw::isFrozen` checks in `DropFunctionStatement` + `DropAggregateStatement` need to be adopted as well.

**Comment by yifan-c** [comment]
> `DescribeStatementTest` has 2 new failures in `testDescribeFunctionAndAggregate` (line 132) + `testDescribe` (line 298)

The tests might be flaky. The last CI run did not have the failures. https://app.circleci.com/pipelines/github/yifan-c/cassandra/77/workflows/5dd298f0-1268-470a-907a-b7f60648dd5e/jobs/391

I think what happened with the failures are caused by pollution from other test cases. The mismatched column fetched is from the other test case. `executeDescribeNet` does pagination. Maybe the page is not advanced.

**Comment by snazy** [comment]
Nope, the two failures in `DescribeStatementTest` were related to the missing changes in DropFunction/AggregateStatement.

**Comment by snazy** [comment]
Committed w/ #686 as a76d288c32e9a062dfafc8e38a33b8aed064913e

**Pull request closed without merging**